protowhat/State.py

```diff
@@ -82,10 +103,15 @@ def get_ast_path(self):
         except StopIteration:
             return self.ast_dispatcher.describe(self.student_ast, "{node_name}")
 
-    def do_test(self, *args, highlight=None, **kwargs):
-        highlight = self.student_ast if highlight is None else highlight
+    def report(self, feedback: Feedback):
+        if feedback.highlight is None and self is not getattr(self, "root_state", None):
```

Comment:
Wait where is "root_state" defined on a `State` object?

Reply:
In depending xwhat libraries, `test_exercise` sets this class property.